src/Layout/Page/Page.component.tsx

```diff
@@ -2,50 +2,215 @@
+};
+
+const DefaultLayoutDefaultProps = {
+    layoutWidth: DEFAULT_LAYOUT_WIDTH,
+    contentWidth: DEFAULT_CONTENT_WIDTH,
+    layoutBackgroundColor: TRANSPARENT,
+    contentBackgroundColor: TRANSPARENT,
+};
+
+const SidebarLayoutDefaultProps = {
+    sidebarWidth: DEFAULT_SIDEBAR_WIDTH,
+    layoutWidth: DEFAULT_CONTENT_WIDTH,
+    contentWidth: DEFAULT_CONTENT_WIDTH,
+    layoutBackgroundColor: TRANSPARENT,
+    contentBackgroundColor: TRANSPARENT,
+};
+
+// Types
+// ------------------------------------------------------------------------------------------------------------------
+type SidebarAlignment = 'left' | 'right';
+
+// Styled Components
+// ------------------------------------------------------------------------------------------------------------------
+const StyledPage = styled.section`
     display: flex;
     flex-direction: column;
     height: 100vh;
-    max-width: 100vw;
-
-    header {
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+    max-width: 100%;
+`;
 
-    footer {
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+const StyledLayout = styled.section<StyledLayoutProps>`
+    max-width: ${props => props.layoutWidth};
+    background-color: ${props => props.layoutBackgroundColor};
+    margin: 0 auto;
 `;
 
-const StyledBodyElement = styled.section`
-    flex: 1 0 auto;
-    background-color: ${colors.silver.base};
-    max-width: 100vw;
-    align-self: stretch;
-    padding: ${sizes.padding.lg} ${sizes.padding.xl};
+const StyledContent = styled.div<StyledContentProps>`
+    max-width: ${props => props.contentWidth};
+    background-color: ${props => props.contentBackgroundColor};
+    margin: 0 auto;
 `;
 
+// Interfaces
+// ------------------------------------------------------------------------------------------------------------------
 interface PageProps {
+    /** The header of the website. This can be a component. */
+    header?: any;
+    /** The footer of the website. This can be a component. */
+    footer?: any;
+    /** The height of the header. It will not hide on overflow. */
+    headerHeight?: string | number;
+    /** The height of the footer. It will not hide on overflow. */
+    footerHeight?: string | number;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface DefaultLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface SidebarLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
```

Comment:
Do are you policing the units here? If you allow only a number then everything can always be in REMs, but it's not quite as intuitive for downstream users. If you allow strings, then people can pass in anything and that can get unwieldy too. I'm not positing a preference, here, but more asking what your thoughts are on this?

Reply:
I had been questioning having string/number as the type almost from the get-go. I think I'm going to yank `number` and keep `string` because ultimately that allows for more flexibility and less confusion.

Reply:
SGTM. I did have Icons implemented the other way, but IIRC Icons now take in a scale prop that's an Enum. Can you do me a favor and check if `Typography` is consistent with this pattern? I think it only takes in a `number` and it assumed that that number is in REMs. If so, can you create a bug ticket and assign it to me to make that consistent with this approach?